lib/assert.js

```diff
@@ -115,7 +116,6 @@ function throwError(message) {
     // the assertion type that failed.
     error.message = `Invalid call to ${stackFrames[2].functionName}() — ` +
       message;
-    error.name = stackFrames[1].functionName.replace(/^Object\./, '');
```

Comment:
`code` isn't a standard property of the `Error` object: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Error

What's the motivation behind not using `name`? If we don't set `name`, all of the exceptions will end up with a `name` of `'Error'`.